src/main/java/appeng/helpers/PatternHelper.java

```diff
@@ -368,17 +382,38 @@ public int hashCode()
 		return this.pattern.hashCode();
 	}
 
-	enum TestStatus
+	@Override
+	public boolean equals( final Object obj )
+	{
+		if( obj == null )
+		{
+			return false;
+		}
+		if( this.getClass() != obj.getClass() )
+		{
+			return false;
+		}
+
+		final PatternHelper other = (PatternHelper) obj;
+
+		if( this.pattern != null && other.pattern != null )
+		{
+			return this.pattern.equals( other.pattern );
+		}
+		return false;
+	}
+
+	private enum TestStatus
 	{
 		ACCEPT, DECLINE, TEST
 	}
 
-	static class TestLookup
+	private final static class TestLookup
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Reorder the modifiers to comply with the Java Language Specification. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AModifiersOrderCheck)